pkg/tsdb/azuremonitor/resourcegraph/azure-resource-graph-datasource.go

```diff
@@ -83,8 +83,7 @@ type argJSONQuery struct {
 }
 
 func (e *AzureResourceGraphDatasource) buildQueries(logger log.Logger, queries []backend.DataQuery, dsInfo types.DatasourceInfo) ([]*AzureResourceGraphQuery, error) {
-	var azureResourceGraphQueries []*AzureResourceGraphQuery
-
+	azureResourceGraphQueries := make([]*AzureResourceGraphQuery, 0, len(queries))
```

Comment:
The Cloud Platform test for the azure monitor is consistently failing in CI. Could you please first merge main? If that doesn't fix the problem, revert the change in this file and we'll see what CI does.

Reply:
we are aware of this issue, it's not related to the changes in this PR. @aangelisc is investigating it atm, will let you know once that is fixed. Sorry for the inconveniences @peakle

Reply:
@joeblubaugh done
@andresmgot ok

Reply:
if you want to unblock this PR you can remove this change so those tests are not triggered. There are usually not that many azure resourge graph queries within a request so the performance improvement here won't be noticeable.

Reply:
@andresmgot good, btw i suppose create another perfomance pr, so i can send it later)